source/common/upstream/upstream_impl.cc

```diff
@@ -118,11 +119,27 @@ ClusterPtr ClusterImplBase::create(const Json::Object& cluster, ClusterManager&
   if (string_type == "static") {
     new_cluster.reset(new StaticClusterImpl(cluster, runtime, stats, ssl_context_manager));
   } else if (string_type == "strict_dns") {
-    new_cluster.reset(new StrictDnsClusterImpl(cluster, runtime, stats, ssl_context_manager,
-                                               dns_resolver, dispatcher));
+    if (cluster.hasObject("dns_resolvers")) {
+      auto resolvers = cluster.getStringArray("dns_resolvers");
```

Comment:
We could consider converting to `Address::Instance` objects here to frontload error checking on address validity (ensuring it looks like "IP:port") and then back to string in `DnsRresolverImpl` constructor. The ares docs are a bit confusing, since they claim that https://c-ares.haxx.se/ares_set_servers_csv.html doesn't fail to bad addresses, so presumably we just start seeing silently failing DNS resolution if we feed in a DNS resolver like "#$!@#JFJ".